### PR TITLE
Re-Add private_cluster_config

### PR DESCRIPTION
**Pull request by hubatish**
After initial PR was [reverted](https://github.com/GoogleCloudPlatform/ai-on-gke/pull/671), re-adding private_cluster-config setting so that public clusters can be created.

The [initial commit](https://github.com/GoogleCloudPlatform/ai-on-gke/commit/039ea4b57edc94086b0aab77cad33394378807b4) had 2 private_cluster_configs defined & clearly didn't work, but I did testing prior to resolving some merge conflicts which I messed up.

Overall goal is allowing public clusters, which are much easier to run kubectl commands with.

### CODE REVIEW

**Comment by hubatish** [comment]
Ran terraform destroy & apply on my existing setup successfully. From blank, terraform apply with private_cluster_config = null in tfvars creates a cluster with "Private cluster" disabled & terraform plan with private_cluster_config = {} from there tries to swap it. Oddly, an already spun up private cluster didn't offer to swap with terraform plan private_cluster_config = null

File: benchmarks/infra/stage-1/README.md [hunk 1]
```diff
@@ -68,27 +68,61 @@ gcloud container fleet memberships get-credentials <cluster-name>
...
+
+| Name | Source | Version |
+|------|--------|---------|
+| <a name="module_gke-infra"></a> [gke-infra](#module\_gke-infra) | ./modules/gke-infra/ | n/a |
+
+## Resources
+
+No resources.
+
+## Inputs
```
Comment (annapendleton) [inline benchmarks/infra/stage-1/README.md:104]: same comment as below

**Review by annapendleton — CHANGES_REQUESTED** [review]

**Comment by annapendleton** [comment]
> Ran terraform destroy & apply on my existing setup successfully. From blank, terraform apply with private_cluster_config = null in tfvars creates a cluster with "Private cluster" disabled & terraform plan with private_cluster_config = {} from there tries to swap it. Oddly, an already spun up private cluster didn't offer to swap with terraform plan private_cluster_config = null

oh interesting - I wonder if we should document this behavior as well? IMO it's likely an implementation detail in the modules we rely on for not changing private back to public

File: benchmarks/infra/stage-1/modules/gke-infra/README.md [hunk 2]
```diff
@@ -73,31 +73,80 @@ module "benchmark-0-infra" {
...
-| [region](variables.tf#L106) | Region used for network resources. | <code>string</code> |  | <code>&#34;us-central1&#34;</code> |
-| [registry_create](variables.tf#L55) | Create remote Docker Artifact Registry. | <code>bool</code> |  | <code>true</code> |
-| [vpc_create](variables.tf#L61) | Project configuration for newly created VPC. Leave null to use existing VPC, or defaults when project creation is required. | <code title="object&#40;&#123;&#10;  name                     &#61; optional&#40;string&#41;&#10;  subnet_name              &#61; optional&#40;string&#41;&#10;  primary_range_nodes      &#61; optional&#40;string, &#34;10.0.0.0&#47;24&#34;&#41;&#10;  secondary_range_pods     &#61; optional&#40;string, &#34;10.16.0.0&#47;20&#34;&#41;&#10;  secondary_range_services &#61; optional&#40;string, &#34;10.32.0.0&#47;24&#34;&#41;&#10;  enable_cloud_nat         &#61; optional&#40;bool, false&#41;&#10;  proxy_only_subnet        &#61; optional&#40;string&#41;&#10;&#125;&#41;">object&#40;&#123;&#8230;&#125;&#41;</code> |  | <code>null</code> |
+<!-- BEGIN_TF_DOCS -->
```
[the remark this answers is not in the document]
Comment (hubatish) [reply benchmarks/infra/stage-1/modules/gke-infra/README.md:76]: Ah, manual modifications on top of running `terraform fmt` is not ideal.. have made those changes though.

File: benchmarks/infra/stage-1/README.md (hunk 1 above)
Comment (hubatish) [reply benchmarks/infra/stage-1/README.md]: Done

**Comment by hubatish** [comment]
> > Ran terraform destroy & apply on my existing setup successfully. From blank, terraform apply with private_cluster_config = null in tfvars creates a cluster with "Private cluster" disabled & terraform plan with private_cluster_config = {} from there tries to swap it. Oddly, an already spun up private cluster didn't offer to swap with terraform plan private_cluster_config = null
> 
> oh interesting - I wonder if we should document this behavior as well? IMO it's likely an implementation detail in the modules we rely on for not changing private back to public

Sure, added a note in the variables.tf description.

File: benchmarks/infra/stage-1/modules/gke-infra/README.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (annapendleton) [reply benchmarks/infra/stage-1/modules/gke-infra/README.md:76]: +1, ty for doing it this time! And we can see if we can find a better long-term solution as well (eg. maybe it makes more sense to slowly shift all our docs to abide by the automated formatted way)

**Review by annapendleton — APPROVED** [review]

**Pull request merged**